app/src/main/java/nl/tudelft/cs4160/trustchain_android/main/OverviewConnectionsActivity.java

```diff
@@ -122,6 +131,36 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
     }
 
+    private void initKey(){
+        KeyPair kp = Key.loadKeys(getApplicationContext());
+        TrustChainDBHelper dbHelper = new TrustChainDBHelper(this);
+        if (kp == null) {
+            kp = Key.createNewKeyPair();
+            Key.saveKey(getApplicationContext(), Key.DEFAULT_PUB_KEY_FILE, kp.getPublic());
+            Key.saveKey(getApplicationContext(), Key.DEFAULT_PRIV_KEY_FILE, kp.getPrivate());
+        }
+        if (isStartedFirstTime(dbHelper, kp)) {
+            MessageProto.TrustChainBlock block = TrustChainBlock.createGenesisBlock(kp);
+            dbHelper.insertInDB(block);
+        }
+    }
+
+    /**
+     * Checks if this is the first time the app is started and returns a boolean value indicating
+     * this state.
+     *
+     * @return state - false if the app has been initialized before, true if first time app started
+     */
+    public boolean isStartedFirstTime(TrustChainDBHelper dbHelper, KeyPair kp) {
+        // check if a genesis block is present in database
+        MessageProto.TrustChainBlock genesisBlock = dbHelper.getBlock(kp.getPublic().getEncoded(), GENESIS_SEQ);
+
+        if (genesisBlock == null) {
```

Comment:
I assume that this is not your code right? I think the  if statement is not needed?

Reply:
Yes good point, but it isn't my code.